servers/rendering/rendering_device.cpp

```diff
@@ -5094,6 +5614,9 @@ Error RenderingDevice::initialize(RenderingContextDriver *p_context, DisplayServ
 		ERR_CONTINUE(err != OK);
 	}
 
+	// TODO: How should this max size be determined?
+	transfer_worker_pool_max_size = OS::get_singleton()->get_processor_count();
```

Comment:
I haven't checked this very deeply yet. So far, I got the impression that this code is not spawning threads but having parallel bookkeeping for thr threads already existing that may be loading resources.

If that's true, a beginning would be `WorkerThreadPool::get_thread_count()`. Even if currently resource loading is only allowed on low priority threads, that can mean the whole thread pool being used because there's the potential of more low-prio tasks in flight than that number and also because the are no pre-assigned low-prio threads, so any thread from the pool can be running a low-prio task at a given moment.

That said, the main thead as well as arbitrary threads can be loading resources at a given point. So, I'd at least add one to that count. If user creates extra threads while a WTP-backed load is in progress, well, it's only fair that they have to wait for a transfer worker to be available.

Reply:
That's correct, it doesn't spawn threads on its own nor does it pre-create them, it just allows a maximum number of concurrent workers. Using WTP as a reference does seem like a good start.